### DISCUSSION TITLE
Pricing of same-region storage transfer

### DISCUSSION THREAD

**Discussion by grschafer** [supabase/supabase#3403] [Questions]
Hello! I've been doing some rough math around the cost of moving some of my infrastructure to supabase and a sticking point is the cost of storage transfer. The supabase pricing page lists $0.09/GB, which matches [S3's pricing](https://aws.amazon.com/s3/pricing/), but a critical piece of S3's pricing is free transfer within the same region:

> You pay for all bandwidth into and out of Amazon S3, except for the following:
> - Data transferred from an Amazon S3 bucket to any AWS service(s) within the same AWS Region as the S3 bucket (including to a different account in the same AWS Region).

I'm wondering -- if I have AWS EC2 instances in the same region as my supabase project and I download from supabase storage to the EC2 instances, will I still be charged $0.09/GB? Or is supabase passing through the AWS cost (which would be $0 or maybe $0.01/GB for EC2 egress?)?

My website gets large file uploads, which are downloaded and processed and stored short-term, so transfer would be the majority of my overall infrastructure costs if it's billed at $0.09/GB.

One potential way to get free same-region transfer is if supabase offered a bring-your-own-bucket option, as has been suggested in https://github.com/supabase/storage-api/issues/27. (That would also let me define an expiration lifecycle rule to remove old files from storage instead of needing to set up a cronjob/workflow, but that's a separate issue from this discussion!)

Another potential solution might be if [supabase/storage-api](https://github.com/supabase/storage-api) ([API docs](https://supabase.github.io/storage-api/#/)) had an endpoint that returned a presigned **S3** url (rather than a supabase.co url), so other AWS infrastructure in the same region could download straight from S3 rather than incurring the transfer costs of `S3 -> storage-api -> downloader`.

Thanks in advance!

**Comment by yadaom** [comment]
I am also looking for the answer to this same question.
Do you have any information on this?

**Reply by GaryAustin1** [reply]
I suggest you ask support your specific pricing question.
